src/main/java/com/yahoo/bullet/postaggregations/ComputationStrategy.java

```diff
@@ -0,0 +1,260 @@
+
+import com.yahoo.bullet.common.BulletError;
+import com.yahoo.bullet.parsing.BinaryExpression;
+import com.yahoo.bullet.parsing.CastExpression;
+import com.yahoo.bullet.parsing.Computation;
+import com.yahoo.bullet.parsing.Expression;
+import com.yahoo.bullet.parsing.LeafExpression;
+import com.yahoo.bullet.parsing.Value;
+import com.yahoo.bullet.record.BulletRecord;
+import com.yahoo.bullet.result.Clip;
+import com.yahoo.bullet.typesystem.Type;
+import com.yahoo.bullet.typesystem.TypedObject;
+import lombok.AllArgsConstructor;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.yahoo.bullet.common.Utilities.extractTypedObject;
+
+@AllArgsConstructor
+public class ComputationStrategy implements PostStrategy {
+    private Computation postAggregation;
+
+    @Override
+    public Clip execute(Clip clip) {
+        clip.getRecords().forEach(r -> {
+                try {
+                    TypedObject result = calculate(postAggregation.getExpression(), r);
+                    switch (result.getType()) {
+                        case INTEGER:
+                            r.setInteger(postAggregation.getNewFieldName(), (Integer) result.getValue());
+                            break;
+                        case LONG:
+                            r.setLong(postAggregation.getNewFieldName(), (Long) result.getValue());
+                            break;
+                        case DOUBLE:
+                            r.setDouble(postAggregation.getNewFieldName(), (Double) result.getValue());
+                            break;
+                        case FLOAT:
+                            r.setFloat(postAggregation.getNewFieldName(), (Float) result.getValue());
+                            break;
+                        case BOOLEAN:
+                            r.setBoolean(postAggregation.getNewFieldName(), (Boolean) result.getValue());
+                            break;
+                        case STRING:
+                            r.setString(postAggregation.getNewFieldName(), (String) result.getValue());
+                            break;
+                        default:
+                            r.setString(postAggregation.getNewFieldName(), "N/A");
+                    }
+                } catch (RuntimeException e) {
+                    r.setString(postAggregation.getNewFieldName(), "N/A");
+                }
+            });
+        return clip;
+    }
+
+    @Override
+    public Optional<List<BulletError>> initialize() {
+        return Optional.empty();
+    }
+
+    private TypedObject calculateLeafExpression(LeafExpression expression, BulletRecord record) {
+        Value value = expression.getValue();
+        TypedObject result = null;
+        String valueString = value.getValue();
+        switch (value.getKind()) {
+            case FIELD:
+                result = extractTypedObject(valueString, record);
+                break;
+            case VALUE:
```

Comment:
I think we should add a type field to Value and use that to treat VALUE as the proper type. We can get rid of kind enum CAST since the presence of type is that.

Then we don't need to infer the types as below. Particularly expecting the usage of ```'``` for strings might not be intuitive inside a JSON string. If not, we should figure out numerics first and default to string so we don't use the ```'```

Reply:
Reminder to change FilterOperations to use forceCast with the type if you do as above.